classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -305,9 +307,42 @@ protected function get_query_actions_sql( array $query, $select_or_count = 'sele
 			$sql          .= " AND a.hook=%s";
 			$sql_params[] = $query[ 'hook' ];
 		}
+
 		if ( ! is_null( $query[ 'args' ] ) ) {
-			$sql          .= " AND a.args=%s";
-			$sql_params[] = $this->get_args_for_query( $query[ 'args' ] );
+			if( isset($query[ 'args_partial_match' ]) && $query[ 'args_partial_match' ] === true && $mysql_version >= '5.7' ) {
+				foreach ( $query[ 'args' ] as $key => $value ) {
+					$supported_types = [ 'integer', 'boolean', 'double', 'string' ];
```

Comment:
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#declaring-arrays
```suggestion
					$supported_types = array( 'integer', 'boolean', 'double', 'string' );
```